src/main/g8/src/main/scala/$package$/Main.scala

```diff
@@ -12,16 +16,20 @@ import $package$.args.Command.{Hi, Pi}
   * @author sbchapin
   * @since 11/11/19.
   */
-object Main {
+object Main extends TaskApp with LazyLogging {
```

Comment:
I like me some TaskApp. The fact that it eliminates the confusion that can often come from combining eagerly-evaluated instances of Future with true monads like Task and that it requires returning an ExitCode are cool and all... but honestly I think the best part is how much more meaningful it makes cats.effect typeclasses like Resource, Ref, MVar, etc.

Reply:
I'd be more eager to accept this conversion if we had TaskApp and the app executions more bridged together in this implementation.  I dislike that when we parse the Args, it just dumps out a unit task rather than dealing with error propagated by incorrectly invoking the args.

Granted, the previous implementation wasn't much better in this regard.

Reply:
That's a good point that I totally missed. I wonder if that's a good reason to throw something like [decline](https://github.com/bkirwi/decline) into this shiz?

I will say this though: If the point of this template is truly to ([as you say](https://github.com/sbchapin/quickstart-scala-sbt-native-image.g8/pull/4#discussion_r747315337)) "establish good and sane practices", I do believe keeping any usages of `unsafeRunSync` out of sight should be among your top priorities.

Reasoning: I found that it's a heck of a lot easier to tell teams (that are just learning functional programming for the first time) to never ever use `unsafeRunSync`, then it is to help settle the arguments that inevitably come from a team finding it and deciding for themselves how often they should use it. 

Maybe that's just my experience though?

Reply:
![in-decline](https://user-images.githubusercontent.com/2286820/141434425-a9411a9e-ec81-45e5-b443-e017dd3a892d.jpg)

For real though, gonna give Decline a try.  I've had minor beef with any arg parsing framework in Scala to-date - maybe this one will be different.

---

> I will say this though: If the point of this template is truly to (as you say) "establish good and sane practices", I do believe keeping any usages of unsafeRunSync out of sight should be among your top priorities.

Agreed.  Let's.

> Reasoning: I found that it's a heck of a lot easier to tell teams (that are just learning functional programming for the first time) to never ever use unsafeRunSync, then it is to help settle the arguments that inevitably come from a team finding it and deciding for themselves how often they should use it.

Often, it's inevitable that someone will make the decision to use unsafeRunSync or its ilk, despite your best intentions.  And it's typically a mismatch between your perceptions of their scope and their actual scope (and their perceptions of that scope). And that's being generous in ruling out bad actors.

I say this because **my immediate reaction is that any sane person would not use unsafeRunSync** _(at least in code expected to be stable and consistent)_.  But I would be wrong.  People get put in tight spots, and they make a sane decision to do this despite what I think.  They do it for a good reason (or at least _good to them_), and a lot of the time, it turns out fine.  My schadenfreude is when it does not, but this isn't "good".  

The productive action is to **educate people on why they shouldn't use unsafeRunSync**, and let him live their life with that knowledge.  I'm thinking something akin to [wat](https://www.destroyallsoftware.com/talks/wat), to illustrate really just what can go wrong.

Reply:
A :fire: meme there lol. Also +1 for usage of the word "schadenfreude". TIL that's a thing.

Reply:
> it's typically a mismatch between your perceptions of their scope and their actual scope (and their perceptions of that scope)

I'm presuming by "scope" here you mean the scope of the problem they are trying to solve. In which case, I can definitely get onboard with that statement.

Reply:
> The productive action is to educate people on why they shouldn't use unsafeRunSync, and let him live their life with that knowledge. I'm thinking something akin to wat, to illustrate really just what can go wrong.

10/10, love that philosophy :100: